### DISCUSSION TITLE
Are there plans for an influxdb v2 addon in the near future?

### DISCUSSION THREAD

**Discussion by ericvb** [hassio-addons/addon-influxdb#113] [General]
I was looking around for moving my current HA Core installation, running in a Python virtual environment, to a more robust production environment. More and more things at home are depending on the HA installation :-)

So I dowloaded the Home Assistant VM (HA OS, HA core and the Supervisor) and imported it in my virtualization environment.
Because I'm running HA Core together with a MariaDb docker image and InfluxDb v2 docker image, I was looking how to move the data.
And here I stumbled on a problem, the current InfluxDb add-on is a 1.x version one.
I don't want to lose my current influxdb data, so this is a problem...

Are there plans for an influxdb v2 addon in the near future?

The move to a more robust system is not urgent, so I can wait if it is in the pipeline.
If not, than I have 2 options:
- install myself InfluxDb v2 as docker image in HA OS, but I don't know if this will be possible without too much changes in this limited linux OS image. I don't want to break future update.
Maybe possible as custom addon, but then I have to read the documentation more thoroughly ;-)
- change course and go for a Home Assistant Supervised installation on a fresh full Ubuntu server.

**Comment by frenck** [comment]
Absolutely! I want to update to InfluxDB 2, however, for me, that is not a top priority right now.

Any help is welcome though 😉

**Reply by ericvb** [reply]
And what kind of help were you thinking?

**Reply by frenck** [reply]
Building it? :D

**Reply by ericvb** [reply]
I'm afraid that this is beyond my knowing.
I suppose it's a little more than forking the repo and adjusting the version number :-)

**Reply by frenck** [reply]
Honestly, I dunno 😄 

But you are right, probably not.

**Reply by ericvb** [reply]
I installed a fresh Home Assistant VM, downloaded your repo and installed it as a local addon.
So I was able to change it to use influxdb2.
As far as I can see, it is running. 
'As far as I can see' = this HA VM has no external sensors yet, so can't send data to the influxdb2 yet, but I can play with the web ui in influxdb2 by accessing it directly by his url http://yourHAVMservername.local:8086.

The part I'm struggling with right now and can't get working is the ingress integration in HA. Lack of knowledge probably.
Influxdb2 includes now Chronograf and Kapacitor. To get the web ui, you use the same port 8086 in the browser.
As far as I can see (again :-) ), it is not possible to set a base path in Influxdb2, so I was redirected to this line in the help of the addons (https://developers.home-assistant.io/docs/add-ons/presentation)

> If the application doesn't support relative paths or you can't set a base url, you can use nginx filter to replace the URL with correct path. Ingress adds a request header X-Ingress-Path that can be used.

Are there examples of this to explain how to use it? Searched on internet, in the HA forum, ... but didn't find an example of using this request header X-Ingress-Path.

**Reply by ericvb** [reply]
See also https://community.home-assistant.io/t/how-to-use-x-ingress-path/276905?u=evb

**Reply by ascheucher** [reply]
Hi Eric,
is there any news regarding your endeavour porting the add on to Influxdb2?
I am curious.

**Reply by frenck** [reply]
Actually I have been playing with the possibility yesterday. Currently I don't see a possible upgrade path for the addon. Thus I might take the path of publishing it as a new addon

**Reply by ericvb** [reply]
@ascheucher I have it working in his own container as local addon, but didn't work further on the ingress integration in HA.
Even with the new documentation examples for ingress integration, I didn't succeed :-(.
Lack of knowledge and no experience in/with nginx, ....

**Reply by ascheucher** [reply]
Thanks for the update!
I am completely new to InfluxDB and Home Assistant. 
I use nginx only as reverse proxy for some webservices I run in docker container on one of our company servers. Maybe I can answer your questions? But don't expect too much, as I just use the basic terminating TSL and routing to the right containers stuff. And it's quite some time ago I set it up. Currently more of a copy and past manoeuvre.

**Reply by ericvb** [reply]
Thanks for the offer to help. It is only already months ago that I played with it.
Maybe it is not necessary if @frenck is making a new addon of it soon  :-)

**Reply by mazzy89** [reply]
@frenck I have some spare time so I can help here to deliver the InfluxDB v2 update add-on. Let me know if you need some hands.

**Reply by ericvb** [reply]
@frenck, is there progress on the roadmap for an Influxdb v2 addon?
It is the only thing stopping me from switching to HomeAssistant OS :-(

**Reply by mazzy89** [reply]
At the end I'm running next to my HA in a separated raspberry pi. Better this way so you can control the many ways to configure influx rather than depends by the add-on

**Reply by lbhm** [reply]
I also tried my luck to get v2 running starting with some minimal changes in a [fork](https://github.com/lbhm/addon-influxdb) of this repo. From what I can tell, InfluxDB itself is running fine but I am also struggling with getting the Ingress integration right :/

@frenck Could you maybe give me/us some hints which changes to the Nginx setup are necessary to get the UI working inside HA? My initial assumption was that since v2 runs on the same port as v1, there should not be many necessary changes.

PS: Please consider my fork as work in progress. I did not update any documentation or config options yet. So far, I just tried to get v2 + Ingress running.

**Reply by jantoney** [reply]
Hi All - Well I've just spent the last 3 days trying to get InfluxDB2 working with NGINX & Ingress and I've resolved to the fact that it's not possible because InfluxDB2 doesn't support any kind of Base URL re-write which is required for how the ingress works.

In light of all this, I've gone and setup my first repository & add-on separate from this one because upgrading a V1 to V2 install is prob going to be too problematic and you can't really add a regression into an add-on by removing support for ingress web UI. Hence - there won't be a pull request to pull it back into this repo.

Anyway - If you're after an InfluxDB2 add-on that doesn't have Ingress Web UI then add this to your Home Assistant repo list, or clone it locally to your own Home Assistant.

Full documentation on how to get it working so your Home Assistant is sending sensor data to it is also in the **Documentation** tab on the add-on.

I've technically marked it as v1.0.0-BETA1 for the time being. Only because I'm the only one that has used it. But I've deleted it and re-added it no less than 30 times and it's working for me. Only tested on AMD64 so far, but can't see why it wouldn't work on ARM/etc as well (another reason for BETA1 tag).
PR's welcome if you find something wrong 😉 

@frenck - I'm going to guess that you wouldn't want to put this into the general community list because it doesn't have Ingress Web UI, but if you do - hit me up and you're welcome to take all the code. I'll keep an eye out for if it becomes possible to do a base URL rewrite then maybe we can chat more.
PS: Hope you don't mind me posting this here. I can see others trying to do their own and thought this might reduce some duplicate work for them

```txt
# URL to add a repo to your Home Assistant (instructions on that page as well)
https://github.com/Jays-Home-Assistant-Add-ons/repository

# URL to check out the add-on and clone locally (if you want to go the hard way)
https://github.com/Jays-Home-Assistant-Add-ons/j-addon-influxdb2
```

**Reply by lbhm** [reply]
Great work @jantoney! I will check out your v2 beta in the next days.

**Reply by ericvb** [reply]
Hi @jantoney, thanks for trying to get this to work :-)

I saw that the normal InfluxDb add-on received a stale as of 2022 maintained flag, meaning that there will be no maintenance any more for the old version of InfluxDb, correct @frenck ?

Meanwhile, 14 days ago, at the beginning of the Christmass holidays, I 'decided/was forced to' to do a reinstall of Home Assistant on new hardware. At the same time I switched from HA Core in a virtual Python environment to HA OS.
I installed InfluxDb v2 as docker image on a virtual machine Ubuntu 20.04 on my NAS, not knowing this...
For the moment I will stick to this setup, but I will follow up this thread :-)

**Reply by jantoney** [reply]
Hey @ericvb - The stale thing doesn't mean it's not maintained, it just means that the add-on hasn't been updated in 2022 (as in no fixes or any updates). Any updates should bring that back into line. It's a way of automatically (somewhat) telling people that it hasn't had an update if someone forgets about their repo and just leaves it.

Let us know on my repo if you have any issues if you decide to deploy v2

**Reply by novisys** [reply]
Hi @frenck 
did you had a sollution for integrating InfluxDB V2 to Hass or that is also not a top priority right now ?
Please help us...

**Reply by ViperRNMC** [reply]
I use this one
https://github.com/Jays-Home-Assistant-Add-ons/j-addon-influxdb2

**Reply by diamant-x** [reply]
I was going to follow this community guide and saw that the official addon is v1 still https://community.home-assistant.io/t/influxdb-2-data-retention-my-approach/457557 . would be great if this can be looked at again. Key requirements for it are: low disk size for long-term storage with downsampling statistical storage.

**Reply by shbatm** [reply]
Revisiting this again for my personal setup--last time I had looked at migrating it was going to be a pain to move all the data from V1 to V2; sometime since it looks like they've upgraded the migration tool https://docs.influxdata.com/influxdb/v2.6/upgrade/v1-to-v2/automatic-upgrade/

However, it doesn't look like the path re-write issue has been resolved to be able to support ingress: https://github.com/influxdata/influxdb/issues/15721

Anyways -- just wanted to drop an update on status.

**Reply by diamant-x** [reply]
I did a manual migration from HA's v1 to official V2 docker, and at the end the 'autonated' script method was to export all current bucket data to a linepoint (compressed!) Per-day collection of files, and import to V2 with retry for any failed day (dua to strange timeouts). It took about 1 minuto or 2 per day so it takes some time to complete but at the end it works.
No ingress, but should be better than keeping an outdated (and possibly with security issues if left unmnantained?) I guess?

**Reply by Dattel** [reply]
> I did a manual migration from HA's v1 to official V2 docker, and at the end the 'autonated' script method was to export all current bucket data to a linepoint (compressed!) Per-day collection of files, and import to V2 with retry for any failed day (dua to strange timeouts). It took about 1 minuto or 2 per day so it takes some time to complete but at the end it works. No ingress, but should be better than keeping an outdated (and possibly with security issues if left unmnantained?) I guess?

I also tried the manual migration way but i used [Jay-Home-Assistant-Add-Ons](https://github.com/Jays-Home-Assistant-Add-ons/j-addon-influxdb2) as template to create my own addon with latest influx2-version and + CLI. Tried to adapt the ingress implementation from the influx1 plugin but failed to get it working with it.

I'm not happy with the current state...so currently i still sticked my live system on official influx. 
you said,you are using the official docker images. is it running on the same machine as homeassistant? how are you planning to update.. commandline? i read, it's not recommended to use other docker images beside the homeassistant installed.

**Comment by Dattel** [comment]
Here is my current playground for the latest influx DB2 implementation. 
it provides latest InfluxDB 2.7.0 + CLI, the latest debian image with s6-overlay..

It can be installed using these repository in homeassistant: https://github.com/Dattel/homeassistant-influxdb2

Would be glad for any feedback or contribution to make that work :-D
Please be kind with me - it's my first try of an plugin for HA

**Reply by michalpulda** [reply]
I guess there is no easy way how to migrate to your add-on from[ Jay's](https://github.com/Jays-Home-Assistant-Add-ons/j-addon-influxdb2), which is sadly not maintained anymore, right? Thanks!

**Reply by Dattel** [reply]
please keep in mind, that my plugin is in early testing.... Feedback is welcome.
i used jay's as inspiration and tried to set up a complete new addon because the way, how services has to be started-up changed a bit...

i assume, you can just install both plugins at the same time and change the port of the second one to e.g. 8087... 
Shutdown both databases and then trying to copy datafiles between these plugins (located in /usr/share/hassio/addons/data/XXXX_influxdb2)

Or you use the build-in influx backup commands using CLI
no quarantee on that... (maybe you have the Jay's version without CLI support?)

BTW: i managed to pull all my data from the official InfluxDB-Addon to my InfluxDB2 using CLI.. Currently my data is getting pumped to both databases -> one using the official influxDB-integration and the other using custom writes through nodered :-D... 
So currently both databases are holding the same informations

**Reply by Dattel** [reply]
Added my first update to 0.0.2
-updates Influx to latest 2.7.1
-updates InfluxCLI to latest 2.7.1
-added a WebURL for easier access

**Reply by mbalasz** [reply]
> i assume, you can just install both plugins at the same time and change the port of the second one to e.g. 8087... Shutdown both databases and then trying to copy datafiles between these plugins (located in /usr/share/hassio/addons/data/XXXX_influxdb2)

I'm considering to upgrade to InfluxDB v2 with your add-on and I saw your post on migrating data. I'm running a Home Assistant OS version on my Raspberry Pi, but I don't have the `/usr/share/hassio/` folder at all. I'm using the SSH addon to check that. I couldn't find any documentation on where the addon data files are stored on a HAOS installation. Do you know if I should check some other locations?

**Reply by Dattel** [reply]
That a bit tricky on hassio and maybe the reason i installed homeassistant as supervised on a debian :-)
With the SSH-Addon you are inside the SSH-Docker container and not on the host.. 
 But i assume, there is a way around.

Have a look here:
https://community.home-assistant.io/t/access-addons-data/428429/3

**Reply by mbalasz** [reply]
Thanks a lot for the pointers. I managed to get the backup following those steps:

1. Follow https://developers.home-assistant.io/docs/operating-system/debugging/ to get debug SSH access to HAOS. Make sure you upload the public key of your other host machine from which you want to SSH.
2. ssh to the host with  `ssh -i <path_to_private_key> root@homeassistant.local -p 22222`
3. Find out the <container_id> of the influxdb container with `docker ps`
4. Create the local backup of influxdb with `docker exec -it <container_id> bash -c 'influxd backup -portable /opt/data'`
5. Copy over the backup directory to local /tmp `docker cp <container-id>:/opt/data /tmp/influxdb/`
6. Get the copy of the data over to your machine via scp `scp -r -P 22222 -O -i <path_to_private_key> root@homeassistant.local:/tmp/influxdb/data/ .`

**Reply by Dattel** [reply]
very nice, that you posted a complete walkthrough here :-D
If i have time, i will include that in the doc and point to this thread :-D

**Reply by Dattel** [reply]
Just want to mention, that there is also a shared directory on "Home Assistant OS"
So with a small adaptation, my initial export/import also works without copy
Just replace @org@ @token@ and @bucket@

```
## Export
docker exec addon_a0d7b954_influxdb \         
  influx_inspect export \
    -datadir "/data/influxdb/data" \
    -waldir "/data/influxdb/wal" \     
    -out "/share/influxdb/export" \                
    -database homeassistant \        
    -retention autogen \  
    -lponly
   
## Import
docker exec addon_ec9cbdb7_influxdb2 \
	influx write \
		--org-id @org@ \
		--token @token@
		--bucket @bucket@ \
		--file "/share/influxdb/export"
```

**Comment by smoki3** [comment]
> Thanks a lot for the pointers. I managed to get the backup following those steps:
> 
> 1. Follow https://developers.home-assistant.io/docs/operating-system/debugging/ to get debug SSH access to HAOS. Make sure you upload the public key of your other host machine from which you want to SSH.
> 2. ssh to the host with  `ssh -i <path_to_private_key> root@homeassistant.local -p 22222`
> 3. Find out the <container_id> of the influxdb container with `docker ps`
> 4. Create the local backup of influxdb with `docker exec -it <container_id> bash -c 'influxd backup -portable /opt/data'`
> 5. Copy over the backup directory to local /tmp `docker cp <container-id>:/opt/data /tmp/influxdb/`
> 6. Get the copy of the data over to your machine via scp `scp -r -P 22222 -O -i <path_to_private_key> root@homeassistant.local:/tmp/influxdb/data/ .`

how do you import then this backup file?
